pybaseball/statcast.py

```diff
@@ -1,175 +1,70 @@
-                retry_success = True
-            else:
-                print(f"Query unsuccessful for data from {end_date} to {end_date}. Skipping these dates.")
-
-            # Reset counter
-            error_counter = 0
-            break
-
-    if dataframe_list:
-        return pd.concat(dataframe_list, axis=0)
-
-    return None
-
-
+@cache.df_cache(expires=365)
 def small_request(start_dt: date, end_dt: date, team: Optional[str] = None, verbose: bool = False) -> pd.DataFrame:
     data = statcast_ds.get_statcast_data_from_csv_url(
         _SC_SMALL_REQUEST.format(start_dt=str(start_dt), end_dt=str(end_dt), team=team if team else '')
     )
     if data is not None and not data.empty:
-        if len(data) == _MAX_SC_RESULTS:
-            if verbose:
-                print('Query completed, but got the maximum number of results. Breaking it down to ensure consistency.')
-            day_span = (end_dt - start_dt).days
-            sub_chunk_1 = small_request(start_dt, start_dt + timedelta(days=day_span // 2), team=team, verbose=verbose)
-            if verbose and sub_chunk_1 is not None and not sub_chunk_1.empty:
-                print(
-                    f"Completed sub-query from {start_dt} to {start_dt + timedelta(days = day_span // 2)} ({len(sub_chunk_1)} results)"
-                )
-            sub_chunk_2 = small_request(
-                start_dt + timedelta(days=day_span // 2 + 1),
-                end_dt,
-                team=team,
-                verbose=verbose
-            )
-            if verbose and sub_chunk_2 is not None and not sub_chunk_2.empty:
-                print(
-                    f"Completed sub-query from {start_dt + timedelta(days = day_span // 2 + 1)} to {end_dt} ({len(sub_chunk_2)} results)"
-                )
-            data = pd.concat([sub_chunk_1, sub_chunk_2], axis=0)
         data = data.sort_values(
             ['game_date', 'game_pk', 'at_bat_number', 'pitch_number'],
             ascending=False
         )
+        if verbose:
+            print(f"Completed sub-query from {start_dt} to {end_dt} ({len(data)} results)")
 
     return data
 
 
-def large_request(start_date: date, end_date: date, step: int, verbose: bool,
+_OVERSIZE_WARNING = '''
+That's a nice request you got there. It'd be a shame if something were to happen to it.
+We strongly recommend that you enable caching before running this. It's as simple as `pybaseball.cache.enable()`.
+Since the Statcast requests can take a *really* long time to run, if something were to happen, like: a disconnect;
+gremlins; computer repair by associates of Rudy Giuliani; electromagnetic interference from metal trash cans; etc.;
+you could lose a lot of progress. Enabling caching will allow you to immediately recover all the successful
+subqueries if that happens.'''
+
+def check_warning(start_dt: date, end_dt: date) -> None:
+    if not cache.config.enabled and (end_dt - start_dt).days >= 42:
```

Comment:
It might be better to think about this in terms of memory rather than days. I could see a case where you trigger this by grabbing the last week of one season and the first week of the next and it still being thrown despite the data covering 2 weeks.

Reply:
Really it should be by ticks of the date range I guess.

Reply:
So I guess I'll switch to this to len() on the range iterator or something. Good call.

Reply:
I'm having second thoughts about this, given that it's a usage warning etc.